### PR TITLE
Paxperscientiam refactor wrapper

### PR DESCRIPTION
**Pull request by paxperscientiam**

### CODE REVIEW

**Comment by paxperscientiam** [comment]
hold off here. need to fix something.

File: src/imagelightbox.js [hunk 1]
```diff
@@ -88,24 +118,7 @@
...
-                }
-                if (options.overlay) {
-                    overlayOff();
-                }
-                if (options.caption) {
-                    captionOff();
-                }
-                if (options.button) {
-                    closeButtonOff();
-                }
```
Comment (rejas) [inline src/imagelightbox.js]: those lines got lost somehow?

File: src/imagelightbox.js [hunk 2]
```diff
@@ -534,7 +537,7 @@
         };
 
         this.quitImageLightbox = function () {
-            quitLightbox();
+            quitImageLightbox();
```
Comment (rejas) [inline src/imagelightbox.js]: why the rename?

File: src/imagelightbox.js [hunk 3]
```diff
@@ -245,9 +246,6 @@
                     return false;
                 });
             },
-            arrowsOff = function () {
-                $('.imagelightbox-arrow').remove();
-            },
```
Comment (rejas) [inline src/imagelightbox.js]: lost too?

File: src/imagelightbox.js [hunk 4]
```diff
@@ -224,17 +230,12 @@
...
             },
             navigationUpdate = function (selector) {
-                var items = $('#imagelightbox-nav').find('a');
+                var items = $navObject.find('a');
                 items.removeClass('active');
                 items.eq($(selector).filter('[href="' + $('#imagelightbox').attr('src') + '"]').index(selector)).addClass('active');
             },
-            navigationOff = function () {
-                $('#imagelightbox-nav').remove();
-            },
```
Comment (rejas) [inline src/imagelightbox.js]: lost?

File: src/imagelightbox.js [hunk 5]
```diff
@@ -163,48 +176,41 @@
...
+                $buttonObject.appendTo($wrapper).on('click', function () {
                     $(this).remove();
-                    quitLightbox();
+                    quitImageLightbox();
                     return false;
                 });
             },
-            closeButtonOff = function () {
-                $('#imagelightbox-close').remove();
-            },
```
Comment (rejas) [inline src/imagelightbox.js]: lost?

File: src/imagelightbox.js [hunk 6]
```diff
@@ -163,48 +176,41 @@
...
+                $activityObject.appendTo($wrapper);
             },
             activityIndicatorOff = function () {
-                $('#imagelightbox-loading').remove();
+                $activityObject.remove();
             },
             overlayOn = function () {
-                $('<div id="imagelightbox-overlay"></div>').appendTo('body');
-            },
-            overlayOff = function () {
```
Comment (rejas) [inline src/imagelightbox.js]: lost?

File: src/imagelightbox.js [hunk 7]
```diff
@@ -1,27 +1,57 @@
...
+            id:'imagelightbox-overlay'
+        }),
+        $navObject = $('<div/>', {
+            id: 'imagelightbox-nav'
+        }),
+        //
+        $wrapper = $('<div/>', {
+            id: 'imagelightbox-wrapper'
+        });
+
```
Comment (rejas) [inline src/imagelightbox.js]: good idea. later we could make those elements configurable. but first this refactor to keep it manageable

**Comment by rejas** [comment]
could you fix those comment issues and rebase it against the latest version? also, make sure that your IDE uses the new .editorconfig file for keeping the code format the same across our repositories.

**Comment by rejas** [comment]
read your comments in #51 , will close this therefor and take a look at that PR

**Comment by paxperscientiam** [comment]
Hey @rejas 

I will look at #52 to make sure I didn't miss anything re the "losses". The reason that code was removed is because -- and correct me if I am wrong -- removing the wrapper should remove all of those other elements as they are appended to the wrapper. I think this is the case with the events.

File: src/imagelightbox.js (hunk 2 above)
Comment (paxperscientiam) [inline src/imagelightbox.js]: Just for (internal) consistency.

File: src/imagelightbox.js (hunk 7 above)
Comment (paxperscientiam) [inline src/imagelightbox.js]: Agreed.

**Pull request closed without merging**